Source/cmake/FindRice.cmake

```
@@ -0,0 +1,139 @@
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#[=======================================================================[.rst:
+FindRice
+--------------
+
+Find Rice headers and libraries.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+``Rice::Proto``
+  The Rice proto library, if found.
+
+``Rice::Io``
+  The Rice io library, if found.
+
+#]=======================================================================]
+
+find_package(PkgConfig QUIET)
+
+if ("Proto" IN_LIST Rice_FIND_COMPONENTS)
+    pkg_check_modules(PC_RICE_PROTO rice-proto)
+    find_path(RiceProto_INCLUDE_DIR
+        NAMES rice-proto.h
+        HINTS ${PC_RICE_PROTO_INCLUDEDIR}
+        ${PC_RICE_PROTO_INCLUDE_DIRS}
+    )
+
+    find_library(Rice_PROTO_LIBRARY
+        NAMES rice-proto
+        HINTS ${PC_RICE_PROTO_LIBDIR} ${PC_RICE_PROTO_LIBRARY_DIRS}
+    )
+
+    if (Rice_PROTO_LIBRARY)
+        set(RiceProto_FOUND TRUE)
+        if (Rice_FIND_REQUIRED_Proto)
+            list(APPEND Rice_LIBS_FOUND "Proto (required): ${Rice_PROTO_LIBRARY}")
+        else ()
+           list(APPEND Rice_LIBS_FOUND "Proto (optional): ${Rice_PROTO_LIBRARY}")
+        endif ()
+    else ()
+        if (Rice_FIND_REQUIRED_Proto)
+           set(_Rice_REQUIRED_LIBS_FOUND OFF)
+           list(APPEND Rice_LIBS_NOT_FOUND "Proto (required)")
+        else ()
+           list(APPEND Rice_LIBS_NOT_FOUND "Proto (optional)")
+        endif ()
+    endif ()
+endif ()
+
+if ("Io" IN_LIST Rice_FIND_COMPONENTS)
+    pkg_check_modules(PC_RICE_IO rice-io)
+    find_path(RiceIo_INCLUDE_DIR
+        NAMES rice-io.h
+        HINTS ${PC_RICE_IO_INCLUDEDIR}
+        ${PC_RICE_IO_INCLUDE_DIRS}
+    )
+
+    find_library(Rice_IO_LIBRARY
+        NAMES rice-io
+        HINTS ${PC_RICE_IO_LIBDIR} ${PC_RICE_IO_LIBRARY_DIRS}
+    )
+
+    if (Rice_IO_LIBRARY)
+        set(RiceIo_FOUND TRUE)
+        if (Rice_FIND_REQUIRED_Io)
+            list(APPEND Rice_LIBS_FOUND "Io (required): ${Rice_IO_LIBRARY}")
+        else ()
+           list(APPEND Rice_LIBS_FOUND "Io (optional): ${Rice_IO_LIBRARY}")
+        endif ()
+    else ()
+        if (Rice_FIND_REQUIRED_Io)
+           set(_Rice_REQUIRED_LIBS_FOUND OFF)
+           list(APPEND Rice_LIBS_NOT_FOUND "Io (required)")
+        else ()
+           list(APPEND Rice_LIBS_NOT_FOUND "Io (optional)")
+        endif ()
+    endif ()
+endif ()
+
+if (NOT Rice_FIND_QUIETLY)
+    if (Rice_LIBS_FOUND)
+        message(STATUS "Found the following Rice libraries:")
+        foreach (found ${Rice_LIBS_FOUND})
+            message(STATUS " ${found}")
+        endforeach ()
+    endif ()
+    if (Rice_LIBS_NOT_FOUND)
+        message(STATUS "The following Rice libraries were not found:")
+        foreach (found ${Rice_LIBS_NOT_FOUND})
+            message(STATUS " ${found}")
+        endforeach ()
+    endif ()
+endif ()
```

Instead of this, we can use the helper macro I mentioned above:


```suggestion
include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(Rice
    VERSION_VAR Rice_VERSION
    HANDLE_COMPONENTS
)
```